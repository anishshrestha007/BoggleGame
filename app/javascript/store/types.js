export const START_GAME = "START_GAME";
export const GET_GAME_DATA = "GET_GAME_DATA";
export const CHECK_WORD = "CHECK_WORD";
export const COMPLETE_GAME = "COMPLETE_GAME";

