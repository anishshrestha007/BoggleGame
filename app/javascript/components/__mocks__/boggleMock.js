export const MOCKDATA = {
    START_GAME: {
        game: {
            userName: 'Anish Shrestha',
            size: 4,
            version: "v1",
            gameTime: 3,
            maxWord: 10,
            minWord:3
        }
    },
    GET_GAME_DATA: {
        game: {
            data: 'asdfghjkloiuytre'
        }
    },

    COMPLETE_GAME: {
        game: {
            isComplete: true
        }
    },
    CHECK_WORD: {
        game: {
            words: [
               [
                {
                    word: 'FACE',
                    score: 2
                  }
               ]
              ],
              totalscore:
                parseInt(0) +
                (isNaN(2) ? 0 : parseInt(2)),
              isCorrect: true,
              totalAttepts: 0 + 1,
              correctAttempts:
                0 + (true ? 1 : 0),
              inCorrectAttempts:
                0 + (true ? 0 : 1)
        }
    },
    


}