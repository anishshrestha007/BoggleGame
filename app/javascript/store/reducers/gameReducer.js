import { GET_THINGS_REQUEST, GET_THINGS_SUCCESS } from "../types";
import config from "../../constants/config";

import { getCurrentUser } from "../../helpers";

export const initialState = {};

const gameReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_THINGS_REQUEST:
      return { ...state, inGame: true, currentUser: action.user, errMsg: [] };
    case GET_THINGS_SUCCESS:
      return { ...state, inGame: true, currentUser: action.user, errMsg: [] };

    default:
      return state;
  }
};

export default gameReducer;
