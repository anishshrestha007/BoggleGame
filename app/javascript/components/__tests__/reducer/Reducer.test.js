import {gameReducer,initialState} from "../../../store/reducers/gameReducer";

import * as types from "../../../store/types";
import { MOCKDATA } from "../../__mocks__/boggleMock";

describe("Boggle reducer test", () => {
  test("should handle start game request", () => {
    expect(
        gameReducer(initialState, {
        type: types.START_GAME,
        game: MOCKDATA.START_GAME.game
      })
    ).toMatchSnapshot();
  });

  test("should handle get game data request", () => {
    expect(
        gameReducer(initialState, {
        type: types.GET_GAME_DATA,
        game: MOCKDATA.GET_GAME_DATA.game
      })
    ).toMatchSnapshot();
  });

  it("should handle check word request", () => {
    expect(
        gameReducer(initialState, {
        type: types.CHECK_WORD,
        game: MOCKDATA.CHECK_WORD.game
      })
    ).toMatchSnapshot();
  });

  it("should handle game complete request", () => {
    expect(
        gameReducer(initialState, {
        type: types.COMPLETE_GAME,
        game: MOCKDATA.COMPLETE_GAME.game
      })
    ).toMatchSnapshot();
  });
});
