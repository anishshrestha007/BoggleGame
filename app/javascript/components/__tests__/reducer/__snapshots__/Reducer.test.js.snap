// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boggle reducer test should handle check word request 1`] = `
Object {
  "gameInfo": Object {
    "correctAttempts": 1,
    "data": null,
    "gameTime": 3,
    "inCorrectAttempts": 0,
    "isComplete": false,
    "isCorrect": true,
    "maxWord": 10,
    "minWord": 3,
    "size": 4,
    "totalAttepts": 1,
    "totalscore": 0,
    "userName": "",
    "version": "v1",
    "words": Array [
      Object {
        "score": undefined,
        "word": undefined,
      },
    ],
  },
}
`;

exports[`Boggle reducer test should handle game complete request 1`] = `
Object {
  "gameInfo": Object {
    "correctAttempts": 0,
    "data": null,
    "gameTime": 3,
    "inCorrectAttempts": 0,
    "isComplete": true,
    "isCorrect": false,
    "maxWord": 10,
    "minWord": 3,
    "size": 4,
    "totalAttepts": 0,
    "totalscore": 0,
    "userName": "",
    "version": "v1",
    "words": Array [],
  },
}
`;

exports[`Boggle reducer test should handle get game data request 1`] = `
Object {
  "gameInfo": Object {
    "correctAttempts": 0,
    "data": undefined,
    "gameTime": 3,
    "inCorrectAttempts": 0,
    "isComplete": false,
    "isCorrect": false,
    "maxWord": 10,
    "minWord": 3,
    "size": 4,
    "totalAttepts": 0,
    "totalscore": 0,
    "userName": "",
    "version": "v1",
    "words": Array [],
  },
}
`;

exports[`Boggle reducer test should handle start game request 1`] = `
Object {
  "gameInfo": Object {
    "correctAttempts": 0,
    "data": null,
    "gameTime": 3,
    "inCorrectAttempts": 0,
    "isComplete": false,
    "isCorrect": false,
    "maxWord": 10,
    "minWord": 3,
    "size": 4,
    "totalAttepts": 0,
    "totalscore": 0,
    "userName": "Anish Shrestha",
    "version": "v1",
    "words": Array [],
  },
}
`;
